File: .cpp/41.FirstMissingPositive.cpp
class Solution
{
public:
    int firstMissingPositive(vector<int> &nums)
    {
        int n = nums.size();
        int i = 0;

        while (i < n)
        {
            int value = nums[i];
            if (value > 0 && value <= n && nums[value - 1] != value)
            {
                swap(nums[i], nums[value - 1]);
            }
            else
            {
                i++;
            }
        }

        for (i = 0; i < n; i++)
        {
            if (nums[i] != i + 1)
            {
                return i + 1;
            }
        }
        return n + 1;
    }
};

File: .py/41.FirstMissingPositive.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        n = len(nums)
        i = 0
        while i<n:
            value = nums[i] - 1

            check = value >= 0 and value < n

            if check and nums[i]!=nums[value]:
                nums[i],nums[value] = nums[value],nums[i]
            else:
                i+=1

        for i in range(n):
            if nums[i]!=i + 1:
                return i + 1

        return n+1